### ISSUE TITLE
test_logfile uselessly calls shutil.rmtree on tearDown

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3208]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3208|
|Type|defect|
|Created|2008-04-25 15:05:10Z|

That causes some intermittent windows failures (http://buildbot.twistedmatrix.com/builders/winxp32-py2.5-select/builds/310/steps/select/logs/problems for example).

Attachments:

* [3208.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bd3/bd33e13f1bcaf70f29a1f1f2f0169928625853f4/64e5eba66a609686e216cbb92d46dc4262c15b94.diff) (1029 bytes) - added by therve on 2008-04-25 15:06:12Z - 
* [3208.2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bd3/bd33e13f1bcaf70f29a1f1f2f0169928625853f4/2e2eb4a260a4f17f5effd8434f408282946cd4a5.diff) (2990 bytes) - added by therve on 2008-04-25 15:16:55Z - 
* [3208.3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bd3/bd33e13f1bcaf70f29a1f1f2f0169928625853f4/ee849e2b098716127b124cf0df8d01452e3f6d22.diff) (3033 bytes) - added by therve on 2008-04-25 15:19:50Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3208 3208
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1209135910000000 1209135910000000
changetime__1209137966000000 1209137966000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Please review!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

The try/finally in `LogFileTestCase.testNoPermission` seems to be unnecessary given `LogFileTestCase.tearDown`.  What do you think?  Can we get rid of that too?

Maybe the comment in `LogFileTestCase.tearDown` could be a docstring instead (and say for _what_ it is necessary).

Basically great, though.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve removed owner|
|-|-|

I got rid of it, but added the chmod on self.dir conditionarly (I don't know if that could create a problem). Docstring done, too.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @therve|
|-|-|

Great.  Thanks.  I only ran the tests on Linux.  I don't see anything that should break on Windows, though.  One other minor change that might be smart.  If somehow we get to `tearDown` and `self.path` exists and `self.dir` is not executable, then `os.path.exists` will return `False`.  Then we'll make the dir readable but the file will still be unreadable.  If we fix the directory first, then this works better.  So flip the order if you agree, then apply.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [23430]) Remove useless shutil.rmtree in test_logfile. This removes some potential
intermittent failures under Windows.

Author: therve
Reviewer: exarkun
Fixes [#3208](https://github.com/twisted/twisted/issues/3208)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
